File: js/tests/capitalizeTheNames.test.js
const capMe = require("../capitalizeTheNames");

const data = [
  [
    ["mavis", "senaida", "letty"],
    ["Mavis", "Senaida", "Letty"]
  ],
  [
    ["samuel", "MABELLE", "letitia", "meridith"],
    ["Samuel", "Mabelle", "Letitia", "Meridith"]
  ],
  [
    ["Slyvia", "Kristal", "Sharilyn", "Calista"],
    ["Slyvia", "Kristal", "Sharilyn", "Calista"]
  ],
  [
    ["krisTopher", "olIva", "herminiA"],
    ["Kristopher", "Oliva", "Herminia"]
  ],
  [
    ["luke", "marsha", "stanford"],
    ["Luke", "Marsha", "Stanford"]
  ],
  [["kara"], ["Kara"]],
  [
    ["mARIANN", "jOI", "gEORGEANN"],
    ["Mariann", "Joi", "Georgeann"]
  ]
];

describe("capMe", () => {
  describe("Return names with only first letter capitalized", () => {
    test.each(data)("capMe(%p) = %p", (arr, result) => {
      let actual = capMe(arr);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/CapitalizeTheNamesTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class CapitalizeTheNamesTest
    {
        [Theory]
        [InlineData(new string[] { "mavis", "senaida", "letty" }, new string[] { "Mavis", "Senaida", "Letty" })]
        [InlineData(new string[] { "samuel", "MABELLE", "letitia", "meridith" }, new string[] { "Samuel", "Mabelle", "Letitia", "Meridith" })]
        [InlineData(new string[] { "Slyvia", "Kristal", "Sharilyn", "Calista" }, new string[] { "Slyvia", "Kristal", "Sharilyn", "Calista" })]
        [InlineData(new string[] { "krisTopher", "olIva", "herminiA" }, new string[] { "Kristopher", "Oliva", "Herminia" })]
        [InlineData(new string[] { "luke", "marsha", "stanford" }, new string[] { "Luke", "Marsha", "Stanford" })]
        [InlineData(new string[] { "kara" }, new string[] { "Kara" })]
        [InlineData(new string[] { "mARIANN", "jOI", "gEORGEANN" }, new string[] { "Mariann", "Joi", "Georgeann" })]
        public void CapMe_NamesStringArray_ReturnNamesOnlyFirstLetterCapitalized(string[] arr, string[] expected)
        {
            var actual = CapitalizeTheNames.CapMe(arr);

            Assert.Equal(expected, actual);
        }
    }
}